recipes/agent_windows_install.rb

```diff
@@ -12,16 +14,10 @@
 
 opts = []
 opts << "/SERVERIP=#{autoregister_values[:go_server_host]}"
+opts << "/S"
 opts << '/D=C:\GoAgent'
 
-if defined?(Chef::Provider::Package::Windows)
-  package 'Go Agent' do
-    source node['gocd']['agent']['package_file']['path']
-    options opts.join(" ")
-  end
-else
-  windows_package 'Go Agent' do
-    source node['gocd']['agent']['package_file']['path']
-    options opts.join(" ")
-  end
+execute "install Go Agent" do
+  command "#{package_path} #{opts.join(' ')}"
+  creates "C:\\GoAgent\\agent.cmd"
```

Comment:
`creates` is a guard here. I had something like this in wrapper cookbook and tested that it does not try to install second time. Why would think it is not idempotent?